clientconn.go

```diff
@@ -966,6 +902,21 @@ func (ac *addrConn) resetTransport(drain bool) error {
 			Addr:     ac.addr.Addr,
 			Metadata: ac.addr.Metadata,
 		}
+		ac.mu.Lock()
+		if ac.state == connectivity.Shutdown {
+			cancel()
+			// ac.tearDown(...) has been invoked.
+			ac.mu.Unlock()
+			return errConnClosing
+		}
+		ac.printf("connecting")
```

Comment:
I would like to keep this in this PR.
We should implement the stats interface using x/net/trace to do tracing like this.